lib/bundler/definition.rb

```diff
@@ -480,14 +482,21 @@ def specs_changed?(source, &block)
         end
       end
 
-      !locked || unlocking || dependencies_for_source_changed?(locked) || source.specs != locked.specs
+      !locked || unlocking || dependencies_for_source_changed?(source) || specs_for_source_changed?(source)
```

Comment:
This change, `dependencies_for_source_changed?(source)` seems to be more about readability.  `source` is the current Source(path, git, etc.) containing the gemspecs from the filesystem for our bundle.  `locked` is the locked Source(path, git, rubygems, etc.) for the same Source as defined in the Gemfile.lock.  `dependencies_for_source_changed?` uses the parameter to find dependencies from our lockfile or gemspecs on the filesystem for that source.  This should have no difference for a source coming from the lockfile or gemspec sources.